app/aqua-behn.hoon

```diff
@@ -0,0 +1,126 @@
+/-  aquarium
+=,  aquarium
+=>  $~  |%
+    +$  move  (pair bone card)
+    +$  card
+      $%  [%poke wire dock %aqua-events (list aqua-event)]
+          [%peer wire dock path]
+          [%pull wire dock ~]
+          [%wait wire p=@da]
+          [%rest wire p=@da]
+      ==
+    ::
+    +$  state
+      $:  %0
+          subscribed=_|
+          piers=(map ship pier)
+      ==
+    ::
+    +$  pier  next-timer=(unit @da)
+    --
+=,  gall
+=|  moves=(list move)
+|_  $:  bowl
+        state
+    ==
+++  this  .
+++  apex  %_(this moves ~)
+++  abet  [(flop moves) this]
+++  emit-moves
+  |=  ms=(list move)
+  %_(this moves (weld ms moves))
+::
+++  emit-aqua-events
+  |=  aes=(list aqua-event)
+  %-  emit-moves
+  [ost %poke /aqua-events [our %aqua] %aqua-events aes]~
+::
+++  poke-aqua-vane-control
+  |=  command=?(%subscribe %unsubscribe)
+  :_  this(subscribed =(command %subscribe))
+  (aqua-vane-control-handler our ost subscribed command)
+::
+++  diff-aqua-effects
+  |=  [way=wire afs=aqua-effects]
+  ^-  (quip move _this)
+  =.  this  apex  =<  abet
+  |-  ^+  this
+  ?~  ufs.afs
+    this
+  =.  this
+    ?+  -.q.i.ufs.afs  this
+      %sleep    abet-pe:handle-sleep:(pe who.afs)
+      %restore  abet-pe:handle-restore:(pe who.afs)
+      %doze     abet-pe:(handle-doze:(pe who.afs) i.ufs.afs)
+    ==
+  $(ufs.afs t.ufs.afs)
+::
+::  Received timer wake
+::
+++  wake
+  |=  [way=wire ~]
+  ^-  (quip move _this)
+  =.  this  apex  =<  abet
+  ?>  ?=([@ *] way)
+  =/  who  (,@p (slav %p i.way))
+  abet-pe:(take-wake:(pe who) t.way ~)
+::
+++  pe
+  |=  who=ship
+  =+  (fall (~(get by piers) who) *pier)
+  =*  pier-data  -
+  |%
+  ++  abet-pe
+    ^+  this
+    =.  piers  (~(put by piers) who pier-data)
+    this
+  ::
+  ++  handle-sleep
+    ^+  ..abet-pe
+    =<  ..abet-pe(pier-data *pier)
+    ?~  next-timer
+      ..abet-pe
+    cancel-timer
+  ::
+  ++  handle-restore
+    ^+  ..abet-pe
+    =.  this
+      %-  emit-aqua-events
+      [%event who [//behn/0v1n.2m9vh %born ~]]~
```

Comment:
Does this need to be the same `/0v1n.2m9vh` as in the ames handler? Maybe that constant should be split off into sur/aquarium/hoon.

Reply:
That value is `u3A->sen`, the "instance number" in vere. It's set to `(scot %uv (mug now))` on boot/restart. I think the idea is to ensure there's a new duct corresponding to the new process, but it's not used for effect routing (or for injecting `%wake` events). I don't think it's needed here. It might not be needed at all, but that's a larger conversation.

Reply:
I just cargo-culted that.  The right answer is probably for aqua to keep track of that, change it on restart, and insert it into each of these wire as necessary.  For now, I've just copied the same string everywhere.